pkg/collector/corechecks/cluster/orchestrator/processors/processor_test.go

```diff
@@ -18,28 +18,103 @@ type Item struct {
 	UID string
 }
 
-func TestChunkItems(t *testing.T) {
-	items := []interface{}{
+func TestChunkOrchestratorMetadataBySizeAndWeight(t *testing.T) {
+	orchestratorMetadata := []interface{}{
 		Item{UID: "1"},
 		Item{UID: "2"},
 		Item{UID: "3"},
 		Item{UID: "4"},
 		Item{UID: "5"},
 	}
-	expected := [][]interface{}{
+	tests := []struct {
+		name                     string
+		maxChunkSize             int
+		maxChunkWeight           int
+		orchestratorMetadata     []interface{}
+		orchestratorMetadataYaml [][]byte
+		expectedChunks           [][]interface{}
+	}{
 		{
-			Item{UID: "1"},
-			Item{UID: "2"},
+			name:                 "chunk by size and weight, one high weight",
+			maxChunkSize:         3,
+			maxChunkWeight:       1000,
+			orchestratorMetadata: orchestratorMetadata,
+			orchestratorMetadataYaml: [][]byte{
+				make([]byte, 1001),
```

Comment:
It is not 100% clear to me, do the `orchestratorMetadata` UID slice order match the `orchestratorMetadataYaml` slice order? If yes, could we make that more clear?

Reply:
Did I understand it correctly, that UID1 is over 1000 and therefore get's its own slice, while 2,3,4 are getting into one due to the maxSize?